shell/lib/asdf.sh

```diff
@@ -40,9 +40,19 @@ asdf_devbase_exec() {
   # shellcheck disable=SC2155
   local tool_env_var="$(echo "$tool" | tr '[:lower:]-' '[:upper:]_')"
 
-  # Why: We're OK with this being the way it is.
-  # shellcheck disable=SC2155
-  local version="$(asdf_get_version_from_devbase "$tool")"
+  # Check if there's a repo level .tool-versions and if so, allow that to override the devbase
+  # version of a specific tool
+  if [[ $tool == "golangci-lint" ]] && grep "golangci-lint" .tool-versions; then
```

Comment:
Can we just use an environment variable instead? This is going to have side-effects in repos, potentially, and having a one-off environment variable would be better.

Will this handle installing the right version if it hasn't already been installed?